applications/eradius_prometheus_collector/src/eradius_prometheus_collector.erl

```diff
@@ -252,18 +258,65 @@ get_metric_info(Name, Stat) ->
         server -> {Metrics, proplists:get_value(Metric, NasFields), RadiusMetricType}
     end.
 
-%% Helper to get a value of a server/nas metric by the given #nas_counter{} index
+%% Helper to get a value of a server/nas metric by the given #nas_counter{} or #client_counter{} index
 %% @private
-get_server_labels_and_val(MetricIdx, #nas_counter{} = Cnt) ->
+get_labels_and_val(_, {#nas_counter{} = Cnt, server}, {Name, Labels})
+  when Name =:= accounting_requests_total;
+       Name =:= accounting_responses_total ->
+    Type = proplists:get_value(acct_type, Labels),
+    {{ServerIP, ServerPort}, NasIP, NasId} = Cnt#nas_counter.key,
+    case lists:member(Type, ?ACCT_TYPES) of
```

Comment:
checking type against ACCT_TYPES is a bit excessive. `acct_type` is either set and valid, or not set. Checking the return for `proplists:get_value` for undefined should be sufficient.

Reply:
I add this checking by reason that the type can be put incorrectly(please, see test point `ok = check_metric_multi({bad_type, EradiusRequestType}, client_accounting_requests_total, [{server_name, good}, {acct_type, bad_type}], 0)`).